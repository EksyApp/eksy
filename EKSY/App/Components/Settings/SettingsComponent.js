import React, {Component} from 'react'
import { View, StyleSheet } from "react-native";
import Header from "../Common/Header";
import FilterSettingsContainer from "../../Containers/Settings/FilterSettingsContainer";
import * as Theme from "../../Theme";
import ProfileContainer from "../../Containers/Settings/ProfileContainer";


export default class SettingsComponent extends Component {
	
	render () {
		return (
				<View style={styles.container}>
					<Header title='Settings' backButton />
					<ProfileContainer/>
					<FilterSettingsContainer/>
				</View>
		)
	}
	
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Theme.backgroundColor,
	}
})