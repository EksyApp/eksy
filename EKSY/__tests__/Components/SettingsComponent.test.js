import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SettingsComponent from "../../App/Components/Settings/SettingsComponent";

jest.mock('react-native-fetch-blob', () => {
	return {
		DocumentDir: () => {},
		polyfill: () => {}
	}
})

jest.mock('react-native-version-number', () => {
    return {
      appversion: 'test'
    }
})

jest.mock('../../App/Containers/Settings/FilterSettingsContainer', () => {
	return require('react-native').View
})

jest.mock('../../App/Containers/Settings/ProfileContainer', () => {
	return require('react-native').View
})

test('SettingsComponent renders correctly', () => {
	const tree = renderer.create(
			<SettingsComponent />
	).toJSON();
	expect(tree).toMatchSnapshot();
});
