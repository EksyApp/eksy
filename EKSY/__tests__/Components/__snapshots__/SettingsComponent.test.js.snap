// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsComponent renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(243,244,229)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(226,227,217)",
        "height": 50,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            Object {
              "height": 50,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 50,
              "zIndex": 1,
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 35,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgb(101,114,122)",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î—„
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(101,114,122)",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Filter Settings
    </Text>
  </View>
  <View />
  <View />
</View>
`;
